mlflow/recipes/steps/ingest/datasets.py

```diff
@@ -519,6 +519,15 @@ class _SparkDatasetMixin:
     and conversion to parquet format.
     """
 
+    def _convert_spark_df_to_pandas(self, spark_df):
+        import pandas as pd
+
+        datetime_cols = [schema[0] for schema in spark_df.dtypes if schema[1].startswith("date")]
```

Comment:
We'd better checking `spark_df.schema` and get the type object from schema field `dataType` attribute, like . `spark_df.schema[column_index].dataType`

then we can the exact DataType object then we can use type matching to check.